functions/assert_metadata.pp

```diff
@@ -25,8 +27,9 @@
 function simplib::assert_metadata (
   String[1] $module_name,
   Optional[Struct[{
-    enable => Optional[Boolean],
-    os     => Optional[Struct[{
+    enable    => Optional[Boolean],
+    blacklist => Optional[Array[String]],
```

Comment:
It looks like `blacklist` can only prohibit entire OSes, but not specific versions.